src/lib/message/L1ToL2Message.ts

```diff
@@ -294,16 +293,34 @@ export class L1ToL2MessageReader extends L1ToL2Message {
 
   /**
    * Receipt for the successful l2 transaction created by this message.
-   * @returns TransactionReceipt of the first successful redeem if exists, otherwise null
+   * @returns TransactionReceipt of the first successful redeem if exists, otherwise null.
+   * Returns expired true if retryable expired without being redeemed, otherwise false.
    */
-  public async getSuccessfulRedeem(): Promise<TransactionReceipt | null> {
+  public async getSuccessfulRedeem(): Promise<
+    | { redeemReceipt: TransactionReceipt | null; expired: false }
+    | { redeemReceipt: null; expired: true }
+  > {
     const l2Network = await getL2Network(this.l2Provider)
     const eventFetcher = new EventFetcher(this.l2Provider)
     const creationReceipt = await this.getRetryableCreationReceipt()
+    if (creationReceipt) {
+      if (!(await this.retryableExists())) {
+        // the retryable was created, but no longer exists
+        // so it must have expired
```

Comment:
not true, they can also be redeemed too

Reply:
It throw `NoTicketWithIDError` instead of 0 for deleted and expired retryable
https://github.com/OffchainLabs/nitro/blob/a2b9435e2cf374df01b277a7a3952ef6542bf81f/precompiles/ArbRetryableTx.go#L136-L138

Reply:
I've updated to return at the end of the function